TypeCobol/Compiler/CodeElements/Entry/DataType.cs

```diff
@@ -23,18 +24,34 @@ public class DataType : ICobolLanguageLevel, IVisitable {
 
 		public override string ToString() { return Name; }
 
-		public override int GetHashCode() { return Name.GetHashCode(); }
 	    public bool AcceptASTVisitor(IASTVisitor astVisitor) {
 	        return astVisitor.Visit(this);
 	    }
 
 	    public override bool Equals(object obj) {
-			var other = obj as DataType;
-			if (other == null) return false;
-			return other == this;
-		}
+            return Equals(obj as DataType);
+        }
+
+        public bool Equals(DataType dataType)
+        {
+            if (Object.ReferenceEquals(this, dataType)) return true;
+            if (Object.ReferenceEquals(null, dataType)) return false;
+
+            return Name.Equals(dataType.Name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 13;
+                hashCode = (hashCode * 397) ^ Name.GetHashCode();
+
+                return hashCode;
+            }
+        }
 
-		public static bool operator ==(DataType x, DataType y)
+        public static bool operator ==(DataType x, DataType y)
```

Comment:
Isn't that declaration supposed to raise an error since we are redefining equality operator on a reference type ? (rule CA1046)

Reply:
When operator equals is overloaded and Equals/GetHashCode are not overrided, CA1046 fires an exception: 'DataType' should not overload the equality operator.
CA1046 is active when Equals/GetHashCode are not overrided.

Reply:
Ok but we should either :
- remove these two operator overloads as this is the only case of operator overloading in the solution and can be confusing. Note that the usages should be all updated.
- or forward the implementation to Equals method to avoid code duplication.
Wait for @smedilol view on this question.

Reply:
OK with @smedilol.